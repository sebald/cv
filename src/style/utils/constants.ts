export const htmlFontSize = 16;

export const ratio = {
  perfectFourth: 1.333,
  perfectFifth: 1.5,
  goldenRatio: 1.618,
  majorSixth: 1.667,
};
