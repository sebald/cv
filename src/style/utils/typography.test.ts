import { htmlFontSize, ratio } from './constants';
import {
  adjustBaseline,
  createScale,
  createTypography,
  getLineHeight,
} from './typography';

test('scale', () => {
  const scale = createScale({
    fontSize: 16,
    htmlFontSize,
    ratio: ratio.perfectFourth,
  });

  expect(scale(1)).toMatchSnapshot();
  expect(scale(2)).toMatchSnapshot();
  expect(scale(3)).toMatchSnapshot();

  expect(scale(0)).toMatchSnapshot();
  expect(scale(-1)).toMatchSnapshot();
});

test('scale (custom)', () => {
  const scale = createScale({
    fontSize: 20,
    htmlFontSize,
    ratio: ratio.goldenRatio,
  });

  expect(scale(1)).toMatchSnapshot();
  expect(scale(2)).toMatchSnapshot();
  expect(scale(3)).toMatchSnapshot();

  expect(scale(0)).toMatchSnapshot();
  expect(scale(-1)).toMatchSnapshot();
});

test('line height', () => {
  expect(
    getLineHeight({ fontSize: '1rem', baseline: 20, htmlFontSize })
  ).toMatchSnapshot();
  expect(
    getLineHeight({ fontSize: 16, baseline: 20, htmlFontSize })
  ).toMatchSnapshot();
  expect(
    getLineHeight({ fontSize: '1rem', baseline: 30, htmlFontSize: 20 })
  ).toMatchSnapshot();

  expect(
    getLineHeight({ fontSize: 20, baseline: 20, htmlFontSize })
  ).toMatchSnapshot();
  expect(
    getLineHeight({ fontSize: 22, baseline: 20, htmlFontSize })
  ).toMatchSnapshot();
});

test('typography', () => {
  const typography = createTypography({
    fontSize: 16,
    baseline: 20,
    htmlFontSize,
    ratio: ratio.goldenRatio,
  });

  expect(typography(0)).toMatchSnapshot();
  expect(typography(1)).toMatchSnapshot();
  expect(typography(2)).toMatchSnapshot();
  expect(typography(-1)).toMatchSnapshot();
});

test('adjust', () => {
  const a1 = adjustBaseline(20);
  expect(a1(5));
  expect(a1(5, 1));

  const a2 = adjustBaseline(12);
  expect(a2(0));
  expect(a2(0, 2));
});
