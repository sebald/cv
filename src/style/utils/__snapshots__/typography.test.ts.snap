// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`line height 1`] = `"1.25rem"`;

exports[`line height 2`] = `"1.25rem"`;

exports[`line height 3`] = `"1.5rem"`;

exports[`line height 4`] = `"1.25rem"`;

exports[`line height 5`] = `"2.5rem"`;

exports[`scale (custom) 1`] = `"2.0225rem"`;

exports[`scale (custom) 2`] = `"3.27241rem"`;

exports[`scale (custom) 3`] = `"5.29475rem"`;

exports[`scale (custom) 4`] = `"1.25rem"`;

exports[`scale (custom) 5`] = `"0.77256rem"`;

exports[`scale 1`] = `"1.333rem"`;

exports[`scale 2`] = `"1.77689rem"`;

exports[`scale 3`] = `"2.36859rem"`;

exports[`scale 4`] = `"1rem"`;

exports[`scale 5`] = `"0.75019rem"`;

exports[`typography 1`] = `
Object {
  "fontSize": "1rem",
  "lineHeight": "1.25rem",
}
`;

exports[`typography 2`] = `
Object {
  "fontSize": "1.618rem",
  "lineHeight": "2.5rem",
}
`;

exports[`typography 3`] = `
Object {
  "fontSize": "2.61792rem",
  "lineHeight": "3.75rem",
}
`;

exports[`typography 4`] = `
Object {
  "fontSize": "0.61805rem",
  "lineHeight": "1.25rem",
}
`;
